service/internal/nodecount/node_count.go

```diff
@@ -28,9 +30,13 @@ func New(c Config) (*NodeCount, error) {
 	if c.K8sClient == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", c)
 	}
+	if c.TenantClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.TenatClient must not be empty", c)
+	}
 
 	nc := &NodeCount{
-		k8sClient: c.K8sClient,
+		k8sClient:    c.K8sClient,
+		tenantClient: c.TenantClient,
```

Comment:
I'm injecting the client here in order to use it here: https://github.com/giantswarm/cluster-operator/pull/1092/files#diff-93dbf7272d7c330881f8e4940096241fR142